drivers/volume/portworx/portworx.go

```diff
@@ -510,174 +514,1520 @@ func (d *portworx) RecoverDriver(n node.Node) error {
+						continue
+					}
+					if tval, ok := nodeVol.Spec.VolumeLabels[matchExp.Key]; ok {
+
+						itval, err := strconv.ParseInt(tval, 10, 64)
+						if err != nil {
+							return false, &ErrFailedToInspectVolume{
+								Cause: fmt.Sprintf("VolumeAntiAffinity with matchExpression Operator:'%v', unable to convert node value to integer, placed on node:%v for volume(%v), having label key:(%v:%v)", matchExp.Operator, node.Id, nodeVol.Id, matchExp.Key, tval),
+							}
+						}
+						mValue, err := strconv.ParseInt(matchExp.Values[0], 10, 64)
+						if err != nil {
+							return false, &ErrFailedToInspectVolume{
+								Cause: fmt.Sprintf("VolumeAntiAffinity with matchExpression Operator:'%v', unable to convert key value(%v) to integer, placed on node:%v for volume(%v),having label key:(%v:%v)", matchExp.Operator, matchExp.Values[0], node.Id, nodeVol.Id, matchExp.Key, tval),
+							}
+						}
+
+						// Check node value is less than spec value
+						if itval <= mValue {
+							return false, &ErrFailedToInspectVolume{
+								Cause: fmt.Sprintf("VolumeAntiAffinity with matchExpression Operator:'%v', is placed on node:%v for volume(%v), having node value(%v) less than Spec value (%v)  for key:(%v:%v)", matchExp.Operator, node.Id, nodeVol.Id, itval, mValue, matchExp.Key, tval),
+							}
+						}
+						logrus.Infof("VolumeAntiAffinity with matchExpression Operator:'%v', is placed on node:%v for volume(%v), having node value(%v) notless than Spec value (%v)  for key:(%v:%v)", matchExp.Operator, node.Id, nodeVol.Id, itval, mValue, matchExp.Key, tval)
+						return true, nil
+
+					}
+				}
+
+				for _, spool := range node.Pools {
+					if tval, ok := spool.Labels[matchExp.Key]; ok {
+						itval, err := strconv.ParseInt(tval, 10, 64)
+						if err != nil {
+							return false, &ErrFailedToInspectVolume{
+								Cause: fmt.Sprintf("VolumeAntiAffinity with matchExpression Operator:'%v', unable to convert node value to integer, placed on node:%v, having label key:(%v:%v)", matchExp.Operator, node.Id, matchExp.Key, tval),
+							}
+						}
+						mValue, err := strconv.ParseInt(matchExp.Values[0], 10, 64)
+						if err != nil {
+							return false, &ErrFailedToInspectVolume{
+								Cause: fmt.Sprintf("VolumeAntiAffinity with matchExpression Operator:'%v', unable to convert key value(%v) to integer, placed on node:%v,having label key:(%v:%v)", matchExp.Operator, matchExp.Values[0], node.Id, matchExp.Key, tval),
+							}
+						}
+
+						// Check node value is less than spec value
+						if itval <= mValue {
+							return false, &ErrFailedToInspectVolume{
+								Cause: fmt.Sprintf("VolumeAntiAffinity with matchExpression Operator:'%v', is placed on node:%v , having node value(%v) less than Spec value (%v)  for key:(%v:%v)", matchExp.Operator, node.Id, itval, mValue, matchExp.Key, tval),
+							}
+						}
+						logrus.Infof("VolumeAntiAffinity with matchExpression Operator:'%v', is placed on node:%v, having n
```

Comment:
overall the logic could be simplified. No need of a second pass here. When you iterate over nodeGrpList above, just make sure the match is within a single node group